docs/dsl_inspec.md

```diff
@@ -252,3 +252,104 @@ control 'ssh-1' do
   end
 end
 ```
+
+# Using Ruby in InSpec
+
+The InSpec DSL is a Ruby based language. This allows you to be flexible with
+Ruby code in controls:
+
+```ruby
+json_obj = json('/file.json')
+json_obj['keys'].each do |value|
+  ..
+end
+```
+
+Ruby allows a lot of freedoms, but should be limited in controls so that they
+remain portable and easy to understand. Please see our [style guide](./style).
+
+Core and custom resources are written as regular Ruby classes which inherit from
+`Inspec.resource`.
+
+
+## Interactive Debugging with Pry
+
+Here's a sample InSpec control that users Ruby variables to instantiate
+an InSpec resource once and use the content in multiple tests.
+
+```ruby
+control 'check-perl' do
+  impact 0.3
+  title 'Check perl compiled options and permissions'
+  perl_out = command('perl -V')
+  #require 'pry'; binding.pry;
+  describe perl_out do
+    its('exit_status') { should eq 0 }
+    its('stdout') { should match (/USE_64_BIT_ALL/) }
```

Comment:
What are your thoughts on:

```
its('stdout') { should match /USE_64_BIT_ALL/ }
```

or

```
its('stdout') { should match(/USE_64_BIT_ALL/) }
```

Reply:
Oh good point! we should probably get that included too 👍